lang/python/python3/patches/020-pass-cross-ranlib-through-distutils.patch

```diff
@@ -0,0 +1,33 @@
+--- a/Lib/distutils/sysconfig.py
++++ b/Lib/distutils/sysconfig.py
+@@ -170,9 +170,9 @@ def customize_compiler(compiler):
+                 _osx_support.customize_compiler(_config_vars)
+                 _config_vars['CUSTOMIZED_OSX_COMPILER'] = 'True'
+ 
+-        (cc, cxx, opt, cflags, ccshared, ldshared, shlib_suffix, ar, ar_flags) = \
++        (cc, cxx, opt, cflags, ccshared, ldshared, shlib_suffix, ar, ranlib, ar_flags) = \
+             get_config_vars('CC', 'CXX', 'OPT', 'CFLAGS',
+-                            'CCSHARED', 'LDSHARED', 'SHLIB_SUFFIX', 'AR', 'ARFLAGS')
++                            'CCSHARED', 'LDSHARED', 'SHLIB_SUFFIX', 'AR', 'RANLIB', 'ARFLAGS')
+ 
+         if 'CC' in os.environ:
+             newcc = os.environ['CC']
+@@ -202,6 +202,8 @@ def customize_compiler(compiler):
+             ldshared = ldshared + ' ' + os.environ['CPPFLAGS']
+         if 'AR' in os.environ:
+             ar = os.environ['AR']
++        if 'RANLIB' in os.environ:
++            ranlib = os.environ['RANLIB']
+         if 'ARFLAGS' in os.environ:
+             archiver = ar + ' ' + os.environ['ARFLAGS']
+         else:
+@@ -215,7 +217,8 @@ def customize_compiler(compiler):
+             compiler_cxx=cxx,
+             linker_so=ldshared,
+             linker_exe=cc,
+-            archiver=archiver)
++            archiver=archiver,
++            ranlib=ranlib)
```

Comment:
Would prefer to find another way to solve this, again unless there is more than one package that requires patching Python directly.

From [this](https://github.com/numpy/numpy/blob/v1.16.3/numpy/distutils/fcompiler/__init__.py#L121) I would assume it looks for a `RANLIB` environment variable?

Reply:
I don't remember all the specifics now but I spent several hours io:stderr:write'ing through the code to hunt down where the ranlib is passed through to the actual executed commands. The code and data flow jumped through files and classes and was very hard to follow.

I've sent this patch to Python, we can wait a bit and see what they think about this (https://bugs.python.org/user32173)

Reply:
the Python thread seems to be a bit slow;
maybe it's vacation time for more people;

let's ping it

Reply:
oh; regarding [the PR](https://github.com/python/cpython/pull/15387), you need to sign a [Python CLA](https://www.python.org/psf/contrib/contrib-form/)

that's some legal stuff that they need, because ¯\_(ツ)_/¯

Reply:
Yeah, I noticed and it is already tagged as having the CLA. The numpy maintainer offered to help with the process if nobody steps up so maybe I should take him up on that offer. :)